### ISSUE TITLE
Diverging implicit expansion for `Show` instances

### ISSUE THREAD

**Issue by vlovgr** [fthomas/refined#553]
The following code (ammonite):

```scala
import $ivy.`eu.timepit::refined-cats:0.9.2`
import cats.implicits._
import eu.timepit.refined.auto._
import eu.timepit.refined.cats._
import eu.timepit.refined.types.string.NonEmptyString

val s: NonEmptyString = "abc"

s.show
```

results in the following compile-time error.

```
diverging implicit expansion for type cats.Contravariant[G]
starting with method refTypeViaMonadError in package cats
```

Changing to `import eu.timepit.refined.cats.refTypeShow` makes the example compile.

**Comment by fthomas** [comment]
Thanks for the report @vlovgr. I've started working on this in https://github.com/fthomas/refined/pull/554.

**Comment by fthomas** [comment]
Fixed by #554
